src/store/auth-context.tsx

```diff
@@ -0,0 +1,132 @@
+import React, { useState } from "react"
+
+type AuthContextObj = {
+  enteredEmail: string
+  enteredPassword: string
+  enterEmail: (value: string) => void
+  enterPassword: (value: string) => void
+  isLogin: boolean
+  handleIsLogin: () => void
+  fetchData: (
+    emailErrorMessage: (status: boolean) => void,
+    passwordErrorMessage: (status: boolean) => void
+  ) => void
+  isLoading: boolean
+  token: undefined | string | null
+  login: (token: string) => void
+  logout: () => void
+}
+
+export const AuthContext = React.createContext<AuthContextObj>({
+  enteredEmail: "",
+  enteredPassword: "",
+  enterEmail: () => {},
+  enterPassword: () => {},
+  isLogin: true,
+  handleIsLogin: () => {},
+  fetchData: () => {},
+  isLoading: false,
+  token: null,
+  login: () => {},
+  logout: () => {},
+})
+
+const AuthContextProvider: React.FC = (props) => {
+  const initialToken = localStorage.getItem('token')
+  const [enteredEmail, setEnteredEmail] = useState<string>("")
+  const [enteredPassword, setEnteredPassword] = useState<string>("")
+  const [isLogin, setIsLogin] = useState<boolean>(true)
+  const [isLoading, setIsLoading] = useState<boolean>(false)
+  const [token, setToken] = useState<string | undefined | null>(initialToken)
+
+  function loginHandler(token: string) {
+    setToken(token)
+    localStorage.setItem('token', token)
+  }
+  function logoutHandler() {
+    setToken(null)
+    setEnteredEmail("")
+    setEnteredPassword("")
+    localStorage.removeItem('token')
+  }
+
+  function enterEmail(value: string) {
+    setEnteredEmail(value)
+  }
+  function enterPassword(value: string) {
+    setEnteredPassword(value)
+  }
+
+  function handleLogin() {
+    setIsLogin((prev) => !prev)
+  }
+
+  let url: string
+  if (isLogin) {
+    url =
+      "https://identitytoolkit.googleapis.com/v1/accounts:signInWithPassword?key=AIzaSyDiGf0rtiDfeXmOmoVyIsZ4AG6aoNZguHY"
```

Comment:
put these urls in config or .env

Reply:
Sure, I`ve created separate config file and added those urls there. With .env - I have some issues with typescript as it didn`t except it`s type (string | undefined), but config works for me.